helix-view/src/view.rs

```diff
@@ -117,6 +117,21 @@ impl View {
         }
     }
 
+    pub fn swap_content(view_a: &mut View, view_b: &mut View) {
+        std::mem::swap(&mut view_a.doc, &mut view_b.doc);
+        std::mem::swap(&mut view_a.offset, &mut view_b.offset);
+        std::mem::swap(&mut view_a.jumps, &mut view_b.jumps);
+        std::mem::swap(
+            &mut view_a.docs_access_history,
+            &mut view_b.docs_access_history,
+        );
+        std::mem::swap(
+            &mut view_a.last_modified_docs,
+            &mut view_b.last_modified_docs,
+        );
+        std::mem::swap(&mut view_a.gutters, &mut view_b.gutters);
+    }
```

Comment:
Rather than doing it this way, wouldn't it be better if you looked up the parent nodes for each view, then swapped the views there? That way there's no need to do all this extra work with `swap_selection_in_views`.

Reply:
It ended up being a little more involved but it's still a lot less work.